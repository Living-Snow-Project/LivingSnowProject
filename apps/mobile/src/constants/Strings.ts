// for TextInput placeholder prop
export const Placeholders = {
  GPS: {
    AcquiringLocation: "Looking for GPS signal",
    EnterCoordinates: "ie. 12.34567, -123.45678",
    NoPermissions:
      "Check location permissions in Settings. Enter coordinates manually.",
    NoLocation: "Could not determine location. Enter coordinates manually.",
  },
  Settings: {
    Username: "Enter your name",
    Organization: "Enter organization you belong to (if any)",
  },
  RecordScreen: {
    LocationDescription: "ie: Blue Lake, North Cascades, WA",
    Notes: "ie. dark red algae on glacial ice",
    Size: "Select a size",
    TubeId: "Leave blank if tube does not have an id",
    TubeIdDisabled: "Disabled - only used for Samples",
  },
};

export const Labels = {
  Date: "Date",
  Delete: "Delete",
  RecordForm: {
    Color: "Color (select all that apply)",
    Gps: "GPS Coordinates (latitude, longitude)",
    Photos: "Select Photos (limit 4)",
    Type: "Are you Reporting a Sighting or Taking a Sample?",
  },
  // TODO: rename to RecordDetailsFields
  RecordFields: {
    Type: "Type",
    Name: "Name",
    Organization: "Organization",
    TubeId: "Tube Id",
    GPSCoordinates: "Coordinates",
    Latitude: "Latitude",
    Longitude: "Longitude",
    Size: "Size",
    Colors: "Colors",
    LocationDescription: "Location Description",
    Notes: "Notes",
    Photos: "Photos",
  },
  TimelineScreen: {
    ExampleRecords: "Example Record",
    PendingRecords: "Pending",
    DownloadedRecords: "Downloaded",
  },
  SettingsScreen: {
    DarkMode: "Dark",
    ManualCoordinates: "Manual Coordinates Warning",
  },
  StatusBar: {
    NoConnection: "No Internet Connection",
  },
  Modal: {
    Cancel: "Cancel",
    Confirm: "Confirm",
    DiskUsage: {
      header: "Confirm Delete",
      body: "Are you sure you want to delete saved photos?",
    },
    GpsManualEntry: {
      header: "Enter coordinates manually?",
      body: `Some users enter coordinates manually if they cannot acquire GPS signal or when they return home.\n\nThis message can be disabled in Settings.`,
    },
  },
};

// for Modal alerts
export const Notifications = {
  uploadSuccess: {
    title: "Record Uploaded",
    message: "Thanks for your submission.",
  },
  uploadRecordFailed: {
    title: "Record Saved",
    message: "We will upload it later.",
  },
  uploadPhotoFailed: {
    title: "Photo Saved",
    message: "We will upload it later.",
  },
  uploadPhotosFailed: {
    title: "Photos Saved",
    message: "We will upload them later.",
  },
  updateRecordSuccess: {
    title: "Record Updated",
  },
  updateRecordFailed: {
    title: "Update Failed",
    message:
      "This can happen when Internet connectivity is restored while updating the record.",
  },
  backgroundTasksNotAllowed: {
    title: "Background Uploads Not Enabled",
    message:
      "Please enable Background App Refresh if you would like stored records to be uploaded in the background.",
  },
};

export const Validations = {
  invalidCoordinates:
    'Coordinates must be in "lat, long" format. eg. 12.34567, -123.45678',
  invalidAlgaeSize: "A size must be specified",
  invalidAlgaeColors: "At least 1 color must be specified",
};

export const Errors = {
  recordNotFound: "Record to update not found",
};

export const RecordDescription = {
  Sample: `Sample`,
  Sighting: `Sighting`,
  Undefined: "Undefined",
};

export const AlgaeSizeDescription = {
  Select: "Select a size",
  Fist: "Fist",
  ShoeBox: "Shoe Box",
  CoffeeTable: "Coffee Table",
  Car: "Car",
  Bus: "Bus",
  Playground: "Playground",
  SportsField: "Sports Field",
  Other: "Other (describe in notes)",
};

export const AlgaeColorDescription = {
  Select: "Select a color",
  Red: "Red",
  Pink: "Pink",
  Grey: "Grey",
  Green: "Green",
  Orange: "Orange",
  Yellow: "Yellow",
  Other: "Other (describe in notes)",
};

export const BackgroundTasks = {
  UploadData: "background-upload-task",
};

// TODO: move this to Labels
export const Headers = {
  DiskUsage: "Disk Usage",
  Profile: "Profile",
  Prompts: "Prompts",
  Theme: "Theme",
};
