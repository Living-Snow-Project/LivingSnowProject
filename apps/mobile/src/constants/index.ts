export * from "./Colors";
export * from "./Strings";
export * from "./TestIds";
