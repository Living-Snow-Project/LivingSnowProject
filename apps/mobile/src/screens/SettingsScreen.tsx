import React, { useState } from "react";
import {
  Box,
  Heading,
  HStack,
  Switch,
  Text,
  VStack,
  useColorMode,
} from "native-base";
import {
  Divider,
  DiskUsage,
  ThemedBox,
  UserIdentityInput,
} from "../components";
import { getAppSettings, setAppSettings } from "../../AppSettings";
import { colorModeManager } from "../providers";
import { Headers, Labels, TestIds } from "../constants";

type SettingsGroupProps = {
  label: string;
  children: JSX.Element;
};

function SettingsGroup({ label, children }: SettingsGroupProps) {
  return (
    <Box px="2">
      <Heading my={1} size="sm">
        {label}
      </Heading>
      {children}
    </Box>
  );
}

type SettingsGroupItemProps = {
  label: string;
  right:
    | JSX.Element
    | ((props: { setLabel: (value: string) => void }) => JSX.Element);
};

function SettingsGroupItem({ label, right }: SettingsGroupItemProps) {
  const [labelValue, setLabel] = useState(label);

  return (
    <HStack height="10" justifyContent="space-between">
      <Text mt="3" fontSize="md">
        {labelValue}
      </Text>
      {typeof right == "function" ? right({ setLabel }) : right}
    </HStack>
  );
}

export function SettingsScreen() {
  const [{ showGpsWarning }, setSettings] = useState(getAppSettings());
  const { colorMode, toggleColorMode } = useColorMode();

  const toggleColorModeAndPersist = () => {
    colorModeManager.set(colorMode == "light" ? "dark" : "light");
    toggleColorMode();
  };

  return (
    <ThemedBox>
      <VStack>
        <Box mb="1">
          <SettingsGroup label={Headers.Profile}>
            <UserIdentityInput />
          </SettingsGroup>
        </Box>
        <Divider />

        <SettingsGroup label={Headers.Prompts}>
          <SettingsGroupItem
            label={Labels.SettingsScreen.ManualCoordinates}
            right={
              <Switch
                testID={TestIds.SettingsScreen.ShowGpsWarning}
                onValueChange={(value) => {
                  setSettings((prev) => ({
                    ...setAppSettings({ ...prev, showGpsWarning: value }),
                  }));
                }}
                isChecked={showGpsWarning}
              />
            }
          />
        </SettingsGroup>
        <Divider />

        <SettingsGroup label={Headers.Theme}>
          <SettingsGroupItem
            label={Labels.SettingsScreen.DarkMode}
            right={
              <Switch
                testID="Dark Mode"
                onValueChange={toggleColorModeAndPersist}
                isChecked={colorMode == "dark"}
              />
            }
          />
        </SettingsGroup>
        <Divider />

        <SettingsGroup label={Headers.DiskUsage}>
          <SettingsGroupItem label="Calculating" right={DiskUsage} />
        </SettingsGroup>
      </VStack>
    </ThemedBox>
  );
}
