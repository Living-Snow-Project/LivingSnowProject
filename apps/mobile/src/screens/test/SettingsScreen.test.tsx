import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { NativeBaseProviderForTesting } from "../../../jesttest.setup";
import { SettingsScreen } from "../SettingsScreen";
import { Placeholders } from "../../constants/Strings";
import { getAppSettings } from "../../../AppSettings";
import { TestIds } from "../../constants/TestIds";

function WrappedSettingsScreen() {
  return (
    <NativeBaseProviderForTesting>
      <SettingsScreen />
    </NativeBaseProviderForTesting>
  );
}
describe("SettingsScreen test suite", () => {
  test("renders", () => {
    const { toJSON } = render(<WrappedSettingsScreen />);

    expect(toJSON()).toMatchSnapshot();
  });

  test("user name", async () => {
    const { getByDisplayValue, getByPlaceholderText } = render(
      <WrappedSettingsScreen />
    );
    const username = getByPlaceholderText(Placeholders.Settings.Username);
    const expected = "Test User Name";

    fireEvent.changeText(username, expected);
    expect(getByDisplayValue(expected)).toBeTruthy();
    expect(getAppSettings().name).toEqual(expected);
    fireEvent(username, "onSubmitEditing");
  });

  test("user organization", () => {
    const { getByDisplayValue, getByPlaceholderText } = render(
      <WrappedSettingsScreen />
    );
    const organization = getByPlaceholderText(
      Placeholders.Settings.Organization
    );
    const expected = "Test Organization";

    fireEvent.changeText(organization, expected);
    expect(getByDisplayValue(expected)).toBeTruthy();
    expect(getAppSettings().organization).toEqual(expected);
    fireEvent(organization, "onSubmitEditing");
  });

  test("toggle showGpsWarning", () => {
    const { getByTestId } = render(<WrappedSettingsScreen />);
    const { showGpsWarning } = getAppSettings();

    fireEvent(getByTestId(TestIds.SettingsScreen.ShowGpsWarning), "onToggle");

    expect(getAppSettings().showGpsWarning).toEqual(!showGpsWarning);
  });
});
