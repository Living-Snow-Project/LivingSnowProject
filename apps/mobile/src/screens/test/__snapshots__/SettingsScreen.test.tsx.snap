// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen test suite renders 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={{}}
  >
    <View
      dataSet={{}}
      style={{}}
    >
      <View
        dataSet={{}}
        style={
          {
            "marginBottom": 4,
          }
        }
      >
        <View
          dataSet={{}}
          style={
            {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <Text
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "color": "#171717",
                "fontFamily": undefined,
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "700",
                "letterSpacing": 0,
                "lineHeight": 20,
                "marginBottom": 4,
                "marginTop": 4,
                "textDecorationLine": undefined,
              }
            }
          >
            Profile
          </Text>
          <View
            dataSet={{}}
            isInvalid={false}
            isRequired={false}
            style={
              {
                "width": "100%",
              }
            }
          >
            <View
              dataSet={{}}
              feedbackId="field-3-feedback"
              hasFeedbackText={true}
              hasHelpText={false}
              helpTextId="field-3-helptext"
              isDisabled={false}
              isInvalid={false}
              isReadOnly={false}
              isRequired={false}
              labelId="field-3-label"
              nativeID="field-3-label"
              setHasFeedbackText={[Function]}
              setHasHelpText={[Function]}
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginBottom": 4,
                  "marginTop": 4,
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#737373",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "500",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Name
              </Text>
            </View>
            <View
              dataSet={{}}
              isFocused={false}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#d4d4d4",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "overflow": "hidden",
                }
              }
            >
              <TextInput
                accessibilityDescribedBy="field-3-feedback"
                accessible={true}
                blurOnSubmit={false}
                dataSet={{}}
                disabled={false}
                editable={true}
                isRequired={false}
                maxLength={50}
                nativeID="field-3-input"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Enter your name"
                placeholderTextColor="#a3a3a3"
                readOnly={false}
                required={false}
                returnKeyType="done"
                secureTextEntry={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  }
                }
              />
            </View>
          </View>
          <View
            dataSet={{}}
            isInvalid={false}
            isRequired={false}
            style={
              {
                "width": "100%",
              }
            }
          >
            <View
              dataSet={{}}
              feedbackId="field-4-feedback"
              hasFeedbackText={true}
              hasHelpText={false}
              helpTextId="field-4-helptext"
              isDisabled={false}
              isInvalid={false}
              isReadOnly={false}
              isRequired={false}
              labelId="field-4-label"
              nativeID="field-4-label"
              setHasFeedbackText={[Function]}
              setHasHelpText={[Function]}
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginBottom": 4,
                  "marginTop": 4,
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#737373",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "500",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Organization
              </Text>
            </View>
            <View
              dataSet={{}}
              isFocused={false}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#d4d4d4",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "overflow": "hidden",
                }
              }
            >
              <TextInput
                accessibilityDescribedBy="field-4-feedback"
                accessible={true}
                blurOnSubmit={true}
                dataSet={{}}
                disabled={false}
                editable={true}
                isRequired={false}
                maxLength={50}
                nativeID="field-4-input"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Enter organization you belong to (if any)"
                placeholderTextColor="#a3a3a3"
                readOnly={false}
                required={false}
                returnKeyType="done"
                secureTextEntry={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        dataSet={{}}
        style={
          {
            "backgroundColor": "#d6d3d1",
            "height": 12,
            "width": "100%",
          }
        }
      />
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        dataSet={{}}
        style={
          {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "700",
              "letterSpacing": 0,
              "lineHeight": 20,
              "marginBottom": 4,
              "marginTop": 4,
              "textDecorationLine": undefined,
            }
          }
        >
          Prompts
        </Text>
        <View
          dataSet={{}}
          style={
            {
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "color": "#171717",
                "fontFamily": undefined,
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 24,
                "marginTop": 12,
                "textDecorationLine": undefined,
              }
            }
          >
            Manual Coordinates Warning
          </Text>
          <View
            dataSet={{}}
            style={{}}
          />
          <RCTSwitch
            accessibilityRole="switch"
            activeThumbColor="#fafafa"
            colorScheme="primary"
            dataSet={{}}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#0891b2"
            style={
              [
                {
                  "height": 31,
                  "width": 51,
                },
                [
                  {},
                  {
                    "backgroundColor": "#d4d4d4",
                    "borderRadius": 16,
                  },
                ],
              ]
            }
            testID="settings-screen-show-gps-warning"
            thumbTintColor="#fafafa"
            tintColor="#d4d4d4"
            value={true}
          />
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        dataSet={{}}
        style={
          {
            "backgroundColor": "#d6d3d1",
            "height": 12,
            "width": "100%",
          }
        }
      />
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        dataSet={{}}
        style={
          {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "700",
              "letterSpacing": 0,
              "lineHeight": 20,
              "marginBottom": 4,
              "marginTop": 4,
              "textDecorationLine": undefined,
            }
          }
        >
          Theme
        </Text>
        <View
          dataSet={{}}
          style={
            {
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "color": "#171717",
                "fontFamily": undefined,
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 24,
                "marginTop": 12,
                "textDecorationLine": undefined,
              }
            }
          >
            Dark
          </Text>
          <View
            dataSet={{}}
            style={{}}
          />
          <RCTSwitch
            accessibilityRole="switch"
            activeThumbColor="#fafafa"
            colorScheme="primary"
            dataSet={{}}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#0891b2"
            style={
              [
                {
                  "height": 31,
                  "width": 51,
                },
                [
                  {},
                  {
                    "backgroundColor": "#d4d4d4",
                    "borderRadius": 16,
                  },
                ],
              ]
            }
            testID="Dark Mode"
            thumbTintColor="#fafafa"
            tintColor="#d4d4d4"
            value={false}
          />
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        dataSet={{}}
        style={
          {
            "backgroundColor": "#d6d3d1",
            "height": 12,
            "width": "100%",
          }
        }
      />
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        dataSet={{}}
        style={
          {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "700",
              "letterSpacing": 0,
              "lineHeight": 20,
              "marginBottom": 4,
              "marginTop": 4,
              "textDecorationLine": undefined,
            }
          }
        >
          Disk Usage
        </Text>
        <View
          dataSet={{}}
          style={
            {
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "color": "#171717",
                "fontFamily": undefined,
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 24,
                "marginTop": 12,
                "textDecorationLine": undefined,
              }
            }
          >
            Calculating
          </Text>
          <View
            dataSet={{}}
            style={{}}
          />
          <View
            dataSet={{}}
            style={{}}
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            colorScheme="primary"
            dataSet={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#0891b2",
                "borderRadius": 4,
                "flexDirection": "row",
                "height": 32,
                "justifyContent": "center",
                "paddingBottom": 4,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 4,
              }
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
              test={true}
            >
              <View
                dataSet={{}}
                style={{}}
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#fafafa",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Delete
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "display": "none",
        },
      ]
    }
  >
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateY": 0,
            },
            {
              "translateX": 0,
            },
            {
              "scale": 1,
            },
            {
              "scaleX": 1,
            },
            {
              "scaleY": 1,
            },
            {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={false}
        collapsable={false}
        dataSet={{}}
        focusable={true}
        importantForAccessibility="no"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "rgb(0, 0, 0)",
            "bottom": 0,
            "left": 0,
            "opacity": 0.3,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateY": 0,
            },
            {
              "translateX": 0,
            },
            {
              "scale": 1,
            },
            {
              "scaleX": 1,
            },
            {
              "scaleY": 1,
            },
            {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View
        closeOnOverlayClick={true}
        dataSet={{}}
        pointerEvents="box-none"
        style={
          {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
        testID="disk-usage-delete-photos-modal"
      >
        <View
          accessibilityViewIsModal={true}
          aria-modal="true"
          dataSet={{}}
          onAccessibilityEscape={[Function]}
          style={
            {
              "backgroundColor": "#fafafa",
              "borderRadius": 8,
              "elevation": 2,
              "maxHeight": 1184,
              "maxWidth": 320,
              "overflow": "hidden",
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 1.41,
              "width": "65%",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            colorScheme="coolGray"
            dataSet={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "transparent",
                "borderRadius": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
                "position": "absolute",
                "right": 12,
                "top": 12,
                "zIndex": 1,
              }
            }
          >
            <RNSVGSvgView
              accessibilityRole="image"
              align="xMidYMid"
              bbHeight="16"
              bbWidth="16"
              dataSet={{}}
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              stroke=""
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "color": "#737373",
                    "height": 16,
                    "width": 16,
                  },
                  {
                    "flex": 0,
                    "height": 16,
                    "width": 16,
                  },
                ]
              }
              tintColor="#737373"
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                propList={
                  [
                    "stroke",
                  ]
                }
                stroke={null}
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M12 9.77778L4.22222 2L2 4.22222L9.77778 12L2 19.7778L4.22222 22L12 14.2222L19.7778 22L22 19.7778L14.2222 12L22 4.22222L19.7778 2L12 9.77778Z"
                    fill={
                      {
                        "type": 2,
                      }
                    }
                    propList={
                      [
                        "fill",
                        "stroke",
                      ]
                    }
                    stroke={null}
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            dataSet={{}}
            style={
              {
                "backgroundColor": "#fafafa",
                "borderBottomWidth": 1,
                "borderColor": "#d4d4d4",
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <Text
              dataSet={{}}
              style={
                {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "600",
                  "letterSpacing": 0,
                  "lineHeight": 20,
                  "textDecorationLine": undefined,
                }
              }
            >
              Confirm Delete
            </Text>
          </View>
          <RCTScrollView
            contentContainerStyle={
              [
                {},
                {
                  "dataSet": {},
                },
                {},
              ]
            }
            dataSet={{}}
            style={{}}
          >
            <View>
              <View
                dataSet={{}}
                style={
                  {
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  }
                }
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Are you sure you want to delete saved photos?
                </Text>
              </View>
            </View>
          </RCTScrollView>
          <View
            dataSet={{}}
            style={
              {
                "backgroundColor": "#fafafa",
                "borderColor": "#d4d4d4",
                "borderTopWidth": 1,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-end",
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="blueGray"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  }
                }
                testID="modal-no-button"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#475569",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Cancel
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 8,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#0891b2",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  }
                }
                testID="modal-confirm-button"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#fafafa",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Confirm
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
